src/fsharp/lexhelp.fs

```diff
@@ -335,14 +337,18 @@ module Keywords =
             match s with 
             | "__SOURCE_DIRECTORY__" ->
                 let filename = fileOfFileIndex lexbuf.StartPos.FileIndex
-                let dirname  = if filename = stdinMockFilename then
-                                   System.IO.Directory.GetCurrentDirectory()
-                               else
-                                   filename |> FileSystem.GetFullPathShim (* asserts that path is already absolute *)
-                                            |> System.IO.Path.GetDirectoryName
+                let dirname  =
+                    if String.IsNullOrWhiteSpace(filename) then
+                        String.Empty
+                    else if filename = stdinMockFilename then
+                        Directory.GetCurrentDirectory()
+                    else
+                        filename
+                        |> FileSystem.GetFullPathShim (* asserts that path is already absolute *)
```

Comment:
Confused by the comment.  The implementation of `GetFullPathShim` has an assert that it's a full path or this call enforces the return is a full path?